src/include/autogen.h.in

```diff
@@ -0,0 +1,49 @@
+/* -*- Mode: C; c-basic-offset:4 ; indent-tabs-mode:nil ; -*- */
+/*
+ *  (C) 2019 by Argonne National Laboratory.
+ *      See COPYRIGHT in top-level directory.
+ */
+
+/* This file will be processed by `maint/gen_init.pl` into autogen.h.
+ * The script will scan through all source file and extracting snippets
+ * into variouls placeholders.
+ *
+ * The main purpose is to gather component init/finalize code so that
+ * a component development do not need to touch code outside its component
+ * tree.
+ *
+ * Potentially, features can be extended.
+ */
+
+/* DISCLAIMER */
+
+#ifndef AUTOGEN_H_INCLUDED
+#define AUTOGEN_H_INCLUDED
+
+/* PLACEHOLDER-DECLARE */
+
+static inline int MPIR_init_components(void)
```

Comment:
Do we really need `static inline` for init time function?

Reply:
Because it is in a header file and I don't think it hurts. But yes, that is not for the performance.

Reply:
OK